e2e-tests/insurance/cypress/e2e/journeys/account/sign-in/enter-code/account-sign-in-create-code-dashboard-or-application-redirect.spec.js

```diff
@@ -0,0 +1,63 @@
+import header from '../../../../../../../partials/header';
+
+import { INSURANCE_ROUTES as ROUTES } from '../../../../../../../constants/routes/insurance';
+import dashboardPage from '../../../../../../../pages/insurance/dashboard';
+
+const {
+  START,
+} = ROUTES;
+
+context('Insurance - Account - Sign in - I want to go straight to my application if there is exactly one application in progress or the dashboard otherwise', () => {
+  before(() => {
+    cy.navigateToUrl(START);
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+  });
+
+  describe('when there are no applications', () => {
+    it('should redirect to the dashboard', () => {
+      cy.completeSignInAndGoToDashboard({});
+      // signs out for next
+      header.navigation.signOut().click();
+    });
+  });
+
+  describe('when there is one application', () => {
+    it('should redirect to the application directly', () => {
+      // creates new application and account
+      cy.completeSignInAndGoToApplication().then((referenceNumber) => {
+        // sign out from application
+        header.navigation.signOut().click();
+
+        // sign in again and complete OTP (without creating account)
+        cy.completeSignInAndOTP({ referenceNumber, shouldRedirectToApplication: true });
+
+        // sign out for next test
+        header.navigation.signOut().click();
+      });
+    });
+  });
+
+  describe('when there is more than one application', () => {
+    it('should redirect to the dashboard', () => {
+      // creates account and application
+      cy.completeSignInAndGoToApplication().then(() => {
+        // go to all applications for user
+        header.navigation.applications().click();
+        // click on start new application page
+        dashboardPage.startNewApplicationButton().click();
+
+        // complete eligibility questions
+        cy.submitInsuranceEligibilityAnswersHappyPath();
+
+        // signs out from application
+        header.navigation.signOut().click();
+
+        // sign in again and check it redirects to dashboard
+        cy.completeSignInAndOTP({ shouldRedirectToApplication: false });
+      });
+    });
```

Comment:
Can we align all the "signs out" comments, could just all be "sign out for next test"?

```
// signs out for next 
// sign out from application
// sign out for next test
// signs out from application
```